src/components/product.js

```diff
@@ -385,16 +368,15 @@ export default class Product extends Component {
    * @return {Array}
    */
   get decoratedOptions() {
-    // const selections = this.selections;
     return this.model.options.map((option) => {
       return {
         name: option.name,
         values: option.values.map((value) => {
           return {
-            name: value,
-            selected: value === option.selected,
-            disabled: false,
-            // disabled: !this.optionValueCanBeSelected(selections, option.name, value),
+            name: value.value, // naming lol
+            selected: Object.values(this.selectedOptions).some((selectedOption) => {
```

Comment:
what's the difference between `.some` and `.filter` and why did we opt for `.some`?